Plug-Ins/NovaTools/System Editor/SystemWindowController.swift

```diff
@@ -0,0 +1,335 @@
+            }
+            // Allow the DataChanged notification to create the view
+            stellar.data = writer.data
+            // Add the stellar to our navdefaults
+            stellarList[destinationNavDefault] = stellar
+
+            // Save the nav defaults
+            let systWriter = BinaryDataWriter()
+            // Make sure there's enough data in the resource to save nav defaults; otherwise initialize default data up to the end of the NavDefaults list
+            if resource.data.count < 2 * 2 {
+                systWriter.write(Int16(0)) // xPos/yPos = 0
+                systWriter.write(Int16(0))
+            }
+            if resource.data.count < (2 + 16 + 16) * 2 {
+                for _ in resource.data.count..<32 {
+                    systWriter.write(Int16(-1)) // no hyperlinks or navdefaults
+                }
+            } else {
+                systWriter.data = resource.data
+            }
+            for (i, navDefault) in stellarList.enumerated() {
+                systWriter.write(Int16(navDefault?.id ?? -1), at: (2 + 16 + i) * 2)
+            }
+            resource.data = systWriter.data
+
+            // Update the view to reflect the new stellar
+            if let view = stellarViews[stellar.id] {
+                view.isHighlighted = true
+                self.syncSelectionFromView(clicked: view)
+            }
+        }
+    }
+
+    // MARK: - Notifications
+
+    @objc func resourceListChanged(_ notification: Notification) {
+        guard let resources = notification.userInfo?["resources"] as? [Resource] else {
+            return
+        }
+        let shouldReload = resources.contains {
+            !$0.data.isEmpty && ["sÿst", "spöb", "spïn", "rlëD", "PICT"].contains($0.typeCode)
+        }
+        if shouldReload {
+            self.reload()
+            if notification.name == .DocumentDidAddResources {
+                // Select added stellars
+                selectedStellars = resources.filter { $0.typeCode == "spöb" }
+                if stellarTable.selectedRow > 0 {
+                    stellarTable.scrollRowToVisible(stellarTable.selectedRow)
+                }
+            }
+        }
+    }
+
+    @objc func resourceIDChanged(_ notification: Notification) {
+        guard let resource = notification.object as? Resource else {
+            return
+        }
+        if ["sÿst", "spöb", "spïn", "rlëD", "PICT"].contains(resource.typeCode) {
+            self.reload()
+        }
+    }
+
+    @objc func resourceNameChanged(_ notification: Notification) {
+        guard let resource = notification.object as? Resource else {
+            return
+        }
+        if resource.typeCode == "sÿst" || resource.typeCode == "spöb" {
+            stellarViews[resource.id]?.updateFrame()
+            if let i = self.row(for: resource) {
+                stellarTable.reloadData(forRowIndexes: [i], columnIndexes: [1])
+            }
+            systemView.needsDisplay = true
```

Comment:
I don't think you need update frames or display the view here?
```suggestion
            if let i = self.row(for: resource) {
                stellarTable.reloadData(forRowIndexes: [i], columnIndexes: [1])
            }
```

Reply:
Looks like that's right, they had no effect so I've removed them.